{"config":{"lang":["en"],"separator":"[\\s\\-]+","pipeline":["stopWordFilter"]},"docs":[{"location":"index.html","title":"Analytics Tasks Guide","text":""},{"location":"index.html#about","title":"About","text":"<p>The project includes 3 main modules.</p> <ul> <li><code>fs</code>: File search to index file text in a flexible way.</li> <li><code>vl</code>: Visual library as a reference point for charts and graphs.</li> <li><code>ao</code>: Automate office to prepares powerpoint presentations using excel .vba, R and python.</li> </ul>"},{"location":"index.html#objectives","title":"Objective(s)","text":"<ol> <li>Text mining.</li> <li>Build PowerPoint presentaitons at scale.</li> <li>Build visualizations with least time and effort.</li> </ol>"},{"location":"file_search.html","title":"File search","text":""},{"location":"file_search.html#about","title":"About","text":""},{"location":"debug/bugfix.html","title":"Bugfix","text":""},{"location":"debug/bugfix.html#bugfix","title":"Bugfix","text":""},{"location":"debug/bugfix.html#automate-office","title":"Automate office","text":""},{"location":"debug/bugfix.html#_template_path-not-found","title":"_template_path not found","text":"Error &amp; Solution Problem: ApplyTemplate<pre><code>*Error comes after running this line\n_control, _template_path = execute_pptx_pipeline(\n    _control,\n    scan_python_functions_from_file_s,\n    _visual_library_dir,\n    _learn_dir,\n    _chart_data_dir,\n    _image_dir,\n    _colors_file,\n    _template_path,\n    _template_pathx,\n    _output_pptm,\n    slide_master_text_elements,\n)\n\n\n*Error Message: \nTraceback (most recent call last):\n  File \"&lt;stdin&gt;\", line 1, in &lt;module&gt;\n  File \"C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\analytics_tasks\\automate_office\\build_batch.py\", line 223, in execute_pptx_pipeline\n    create_or_apply_potm(_template_pathx, outputpptm, _control)\n  File \"C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\analytics_tasks\\automate_office\\build_batch.py\", line 1508, in create_or_apply_potm\n    ppt.ApplyTemplate(_template_path)\n  File \"C:\\Users\\user\\AppData\\Local\\Temp\\gen_py\\3.11\\91493440-5A91-11CF-8700-00AA0060263Bx0x2x12\\_Presentation.py\", line 50, in ApplyTemplate\n    return self.oleobj.InvokeTypes(2007, LCID, 1, (24, 0), ((8, 1),),FileName\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\npywintypes.com_error: (-2147352567, 'Exception occurred.', (0, 'Microsoft PowerPoint', 'Presentation.ApplyTemplate : Object does not exist.', '', 0, -2147188720), None)\n\n\n*Solution: Complete win32 postinstall\ncd C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\npython Scripts/pywin32_postinstall.py -install\n\n\n*Powershell message postinstall\nParsed arguments are: Namespace(install=True, remove=False, wait=None, silent=False, quiet=False, destination='C:\\\\Users\\\\user\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python311\\\\Lib\\\\site-packages')\nCopied pythoncom311.dll to C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\\pythoncom311.dll\nCopied pywintypes311.dll to C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\\pywintypes311.dll\nYou do not have the permissions to install COM objects.\nThe sample COM objects were not registered.\n-&gt; Software\\Python\\PythonCore\\3.11\\Help[None]=None\n-&gt; Software\\Python\\PythonCore\\3.11\\Help\\Pythonwin Reference[None]='C:\\\\Users\\\\user\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python311\\\\Lib\\\\site-packages\\\\PyWin32.chm' \nRegistered help file\nPythonwin has been registered in context menu\nCreating directory C:\\Users\\user\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\win32com\\gen_py\nShortcut for Pythonwin created\nShortcut to documentation created\nThe pywin32 extensions were successfully installed.\n</code></pre>"},{"location":"debug/bugfix.html#file-search","title":"File search","text":""},{"location":"debug/bugfix.html#file-search-xlrd-issue","title":"File search xlrd issue","text":"Error &amp; Solution Problem: xlrd not installed<pre><code>*Error comes after running this line\nscan_drives(scan, scan_ext)\n\n*Error message\nc:\\time_series\\example\\exponantial_smoothing.xls\n2025-06-24 19:57:04,540 | [INFO] | user | ERROR: function load_ifp_xlsx... cannot read file c:\\time_series\\example\\exponantial_smoothing.xls\n2025-06-24 19:57:04,540 | [INFO] | user | Error details: openpyxl does not support the old .xls file format, please use xlrd to read this file, or convert it to the more recent .xlsx file format.\n\n*Solution\nuv pip install --system xlrd\n</code></pre>"}]}